pkgs/development/libraries/mozc/default.nix

```
@@ -0,0 +1,235 @@
+      owner = "musjj";
+      repo = "jp-zip-codes";
+      name = repo;
+      rev = "be9cd2ac888a34eba30a425c43a7e54f5187a649";
+      hash = "sha256-02CE/DJnqNXUF0OH3ptq2sb8UbtqKjLKx7Hf0izq+X4=";
+    })
+  ];
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [
+    glibcLocales
+    pkg-config
+    python3
+    ruby
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [ qtbase ];
+
+  preBuild = ''
+    cd mozc/src
+  '';
+
+  LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive";
+  LC_ALL = "en_US.UTF-8";
+
+  inherit bazel;
+  removeRulesCC = false;
+  dontAddBazelOpts = true;
+
+  bazelFlags = [
+    "--config"
+    "oss_linux"
+    "--compilation_mode"
+    "opt"
+  ];
+
+  bazelTargets = [
+    "server:mozc_server"
+    "gui/tool:mozc_tool"
+  ];
+
+  fetchAttrs = {
+    postPatch = ''
+      substituteInPlace mozc/src/WORKSPACE.bazel \
+        --replace \
+          'url = "https://www.post.japanpost.jp/zipcode/dl/kogaki/zip/ken_all.zip"' \
+          "url = \"file://$PWD/jp-zip-codes/ken_all.zip\"" \
+        --replace \
+          'url = "https://www.post.japanpost.jp/zipcode/dl/jigyosyo/zip/jigyosyo.zip"' \
+          "url = \"file://$PWD/jp-zip-codes/jigyosyo.zip\""
+    '';
+
+    sha256 = {
+      x86_64-linux = "sha256-6Md44d/TBD62OjCmrWT9s+Y/EAdKGhhLYL9pp2ryOQo=";
+      aarch64-linux = "sha256-B2KpzgOaUoojtg8SHZUH4Hjkrw8wSdCFmZqTtEK28eY=";
+    }.${stdenv.system};
+  };
+
+  buildAttrs = {
+    postPatch = ''
+      sed -ri -e "s|^(LINUX_MOZC_SERVER_DIR = ).+|\1\"$out/lib/mozc\"|" mozc/src/config.bzl
+
+      (
+        cd merge-ut-dictionaries/src
+
+        sed -i -e "s|https://raw.githubusercontent.com/google/mozc/master|../../mozc|" remove_duplicate_ut_entries.rb
+
+        # Use a dated snapshot for the JAWiki dump data
+        sed -i -e '/wget/d' count_word_hits.rb
+        sed -i -e "s|^filename = \"jawiki-.*|filename = \"../../jawiki/jawiki.gz\"|" count_word_hits.rb
+
+        [[ -e mozcdic-ut.txt ]] && rm mozcdic-ut.txt
+
+        dicts=(
+          alt-cannadic
+          edict2
+          jawiki
+          neologd
+          personal-names
+          place-names
+          skk-jisyo
+          sudachidict
+        )
+
+        for dict in "''${dicts[@]}"; do
+          tar -xf ../../mozcdic-ut-$dict/mozcdic-ut-$dict.txt.tar.bz2
+          cat mozcdic-ut-$dict.txt >>mozcdic-ut.txt
+        done
+
+        ruby remove_duplicate_ut_entries.rb mozcdic-ut.txt
+        ruby count_word_hits.rb
+        ruby apply_word_hits.rb mozcdic-ut.txt
+
+        cat mozcdic-ut.txt >>../../mozc/src/data/dictionary_oss/dictionary00.txt
+      )
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/share/licenses/mozc
+      cp ../LICENSE $out/share/licenses/mozc/LICENSE
+      cp data/installer/credits_en.html $out/share/licenses/mozc/Submodules
+
+      mkdir -p $out/lib/mozc
+      cp bazel-bin/server/mozc_server $out/lib/mozc/mozc_server
+      cp bazel-bin/gui/tool/mozc_tool $out/lib/mozc/mozc_tool
+
```

```suggestion
      mkdir -p $out/bin
      cp bazel-bin/unix/emacs/mozc_emacs_helper $out/bin/mozc_emacs_helper

```
Install `mozc_emacs_helper`.